### ISSUE TITLE
Inconsistent behavior

### ISSUE THREAD

**Issue by carletex** [BuidlGuidl/PWA-burner-wallet#3]
I deployed it here: https://basic-pwa.vercel.app/

It worked for me initially.. and for Austin too:

![photo1696435408](https://github.com/BuidlGuidl/PWA-burner-wallet/assets/2486142/84ad2ac3-1ab1-47ca-bec8-8c7de9d9af09)

But now I get this:
![photo1696439330](https://github.com/BuidlGuidl/PWA-burner-wallet/assets/2486142/01b02fbb-7586-4a74-9602-4e777abd086b)

Austin also got an error the first time
![photo1696435412](https://github.com/BuidlGuidl/PWA-burner-wallet/assets/2486142/3efb6637-9065-4d7b-bab7-e158a8d1a88e)

![image](https://github.com/BuidlGuidl/PWA-burner-wallet/assets/2486142/3da18d08-84cb-4768-aad8-90302df493d1)

----

Any ideas on how to debug this in a good way?

cc @technophile-04 @Pabl0cks

**Comment by technophile-04** [comment]
Can confirm this inconsistent on mobile :(, happened to me many time 

For some reason `navigator.serviceWorker.getRegistration()` and `swRegistration.pushManager.subscribe(options)` from https://github.com/BuidlGuidl/PWA-burner-wallet/blob/b17797cea25b39efba00010244b217f66d5e6521/packages/nextjs/utils/service-workers/index.ts#L12

is not working properly on mobile devices, I am on it and will try to fix this nicely 🙌

**Comment by carletex** [comment]
Closing this too.

But @Pabl0cks , feel free to do a test again with the merged changes from #4 and see if your issues are solved. We can open more specific issues if not.

**Comment by Pabl0cks** [comment]
Awesome job @technophile-04 with the last PR this is working great!! 🔥🙌 

This is what I've found (very edge cases probably):
- **Environment variables on Vercel only work after a Vercel deployment**, so if you `yarn vercel` from console, then set the variables when the project is created in Vercel, you'll need to `yarn vercel` again to make push notifications work.  
I think this is working as intended due to [Vercel specs](https://github.com/vercel/vercel/discussions/7728), and just need to be documented in Readme, I'll take care of that 🙌

- Sometimes after a period of inactivity, my first *"Notify all"* throws a `500 internal error`. Refreshing the webpage fixed it, and happened only like once or twice. I think we can leave it for now probably.

- After some usage in several browsers on desktop and mobile (like 3 on each one), sometimes either my Desktop (all browsers) or my Mobile (all browsers) stopped getting notifications, but the other device still was receiving them. No errors, everything OK in console.  
Tried cleaning firestore database, tried giving it time to maybe "unban" my device if was some kind of spam problem, also tried re-deploying and generating new keys. Nothing worked to try to fix it, **until I tried using a VPN, suddenly all my unreceived notifications started popping**. Seems like either Vercel, Firebase or your own device is preventing spam and not showing notifications on suspicious devices.
**Probably this doesn't happend with normal usage**

TYSM Shiv for the fast fixes, will try to understand everything you changed 🙃

**Comment by technophile-04** [comment]
>  Seems like either Vercel, Firebase or your own device is preventing spam and not showing notifications on suspicious devices.

Yeah __I think__ its mostly because of our own browser tries to safeguard users from spam notifications but not 100% sure, will try to have a look and will also have a look at pt.2 if I reproduces for me  🙌

but tysm for the great testing !!!!
